packages/design-system/src/components/tooltip/index.js

```diff
@@ -290,41 +341,58 @@ function Tooltip({
-        placement={dynamicPlacement}
-        spacing={spacing}
-        isOpen={Boolean(shown && (shortcut || title))}
-        onPositionUpdate={positionArrow}
-        zIndex={popupZIndexOverride}
-        noOverFlow
-        ignoreMaxOffsetY={ignoreMaxOffsetY}
-      >
-        <TooltipContainer
-          className={className}
-          ref={tooltipRef}
-          placement={dynamicPlacement}
-          shown={shown}
-          {...tooltipProps}
-        >
-          <TooltipText size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.X_SMALL}>
-            {shortcut ? `${title} (${prettifyShortcut(shortcut)})` : title}
-          </TooltipText>
-          {hasTail && (
-            <>
-              <SvgForTail>
-                <clipPath
-                  id={SVG_TOOLTIP_TAIL_ID}
-                  clipPathUnits="objectBoundingBox"
+      {popupState?.offset && isOpen
+        ? createPortal(
+            <PopupContainer
+              ref={popup}
+              $offset={
+                dynamicOffset
+                  ? {
+                      ...popupState.offset,
+                      x: popupState.offset.x + dynamicOffset?.x,
+                    }
+                  : popupState.offset
+              }
+              noOverFlow
+              zIndex={popupZIndexOverride || DEFAULT_POPUP_Z_INDEX}
+              transforms={getTransforms(dynamicPlacement, isRTL)}
+            >
+              <TooltipContainer
+                className={className}
+                ref={tooltipRef}
+                shown={shown}
+                zIndex={popupZIndexOverride || DEFAULT_POPUP_Z_INDEX}
+                {...styleOverride}
+              >
+                <TooltipText
+                  size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.X_SMALL}
                 >
-                  <path d="M1,1 L0.868,1 C0.792,1,0.72,0.853,0.676,0.606 L0.585,0.098 C0.562,-0.033,0.513,-0.033,0.489,0.098 L0.399,0.606 C0.355,0.853,0.283,1,0.207,1 L0,1 L1,1" />
-                </clipPath>
-              </SvgForTail>
-              <Tail placement={dynamicPlacement} translateX={arrowDelta} />
-            </>
-          )}
-        </TooltipContainer>
-      </Popup>
+                  {shortcut
+                    ? `${title} (${prettifyShortcut(shortcut)})`
+                    : title}
+                </TooltipText>
+                {hasTail && (
+                  <>
+                    <SvgForTail>
+                      <clipPath
+                        id={SVG_TOOLTIP_TAIL_ID}
+                        clipPathUnits="objectBoundingBox"
+                      >
+                        <path d="M1,1 L0.868,1 C0.792,1,0.72,0.853,0.676,0.606 L0.585,0.098 C0.562,-0.033,0.513,-0.033,0.489,0.098 L0.399,0.606 C0.355,0.853,0.283,1,0.207,1 L0,1 L1,1" />
+                      </clipPath>
+                    </SvgForTail>
+                    <Tail
+                      placement={dynamicPlacement}
+                      translateX={-dynamicOffset?.x || arrowDelta}
```

Comment:
This needs to be opposite of dynamicOffset since when to push the arrow back into the tooltip.